chain/src/main/scala/org/bitcoins/chain/models/BlockHeaderDAO.scala

```diff
@@ -339,14 +339,38 @@ case class BlockHeaderDAO()(implicit
       ec: ExecutionContext): Future[Vector[Blockchain]] = {
     getBetweenHeights(from = from, to = to).map { headers =>
       if (headers.map(_.height).distinct.size == headers.size) {
```

Comment:
Some preliminary optimziations as `getBlockchainsBetweenHeights()` seems like it can have some pretty pathogical behavior in the case of forks (especially large ranges of heights!) 

```scala

  @tailrec
  private def loop(
      chains: Vector[Blockchain],
      allHeaders: Vector[BlockHeaderDb]): Vector[Blockchain] = {
    val usedHeaders = chains.flatMap(_.headers).distinct
    val diff = allHeaders.filter(header =>
      !usedHeaders.exists(_.hashBE == header.hashBE))
    if (diff.isEmpty) {
      chains
    } else {
      val sortedDiff = diff.sortBy(_.height).reverse

      val newChainHeaders =
        Blockchain.connectWalkBackwards(sortedDiff.head, allHeaders)
      val newChain = Blockchain(newChainHeaders.sortBy(_.height).reverse)
      loop(chains :+ newChain, allHeaders)
    }
  }

  /** Retrieves a blockchain with the best tip being the given header */
  def getBlockchainsBetweenHeights(from: Int, to: Int)(implicit
      ec: ExecutionContext): Future[Vector[Blockchain]] = {
    getBetweenHeights(from = from, to = to).map { headers =>
      val distinctHeights = headers.map(_.height).distinct
      if (distinctHeights.size == headers.size) {
        val headersByHeight = headers.sortBy(_.height)
        Vector(Blockchain.fromHeaders(headersByHeight.reverse))
      } else {
        val headersByHeight: Vector[(Int, Vector[BlockHeaderDb])] =
          headers.groupBy(_.height).toVector
        val tips: Vector[BlockHeaderDb] = {
          headersByHeight.sortBy(_._1).last._2
        }
        val allHeaders = headers

        val chains = tips.map { tip =>
          Blockchain
            .connectWalkBackwards(tip, allHeaders)
            .sortBy(_.height)
            .reverse
        }
        val init = chains.map(Blockchain(_)).distinct

        loop(init, allHeaders).distinct
      }
    }
  }
```

Reply:
Applied these plus some others